src/main/java/gregtech/common/metatileentities/storage/MetaTileEntityQuantumChest.java

```diff
@@ -122,43 +128,35 @@ public void update() {
                 ItemStack outputStack = exportItems.getStackInSlot(0);
-                if (outputStack.isEmpty() || outputStack.isItemEqual(inputStack) && ItemStack.areItemStackTagsEqual(inputStack, outputStack)) {
-                    if (!inputStack.isEmpty() && (virtualItemStack.isEmpty() || areItemStackIdentical(virtualItemStack, inputStack))) {
-                        int amountOfItemsToInsert = (int) Math.min(inputStack.getCount(), maxStoredItems - itemsStoredInside);
-                        if (this.itemsStoredInside == 0L || virtualItemStack.isEmpty()) {
-                            this.virtualItemStack = GTUtility.copy(1, inputStack);
-                        }
-                        inputStack.shrink(amountOfItemsToInsert);
-                        importItems.setStackInSlot(0, inputStack);
-                        this.itemsStoredInside += amountOfItemsToInsert;
-
-                        markDirty();
-                    }
+                if (!inputStack.isEmpty() && (virtualItemStack.isEmpty() || areItemStackIdentical(outputStack, inputStack))) {
+                    GTTransferUtils.moveInventoryItems(importItems, combinedInventory);
+
+                    markDirty();
                 }
             }
             if (itemsStoredInside > 0 && !virtualItemStack.isEmpty()) {
                 ItemStack outputStack = exportItems.getStackInSlot(0);
                 int maxStackSize = virtualItemStack.getMaxStackSize();
                 if (outputStack.isEmpty() || (areItemStackIdentical(virtualItemStack, outputStack) && outputStack.getCount() < maxStackSize)) {
-                    int amountOfItemsToRemove = (int) Math.min(maxStackSize - outputStack.getCount(), itemsStoredInside);
-                    if (outputStack.isEmpty()) {
-                        outputStack = GTUtility.copy(amountOfItemsToRemove, virtualItemStack);
-                    } else outputStack.grow(amountOfItemsToRemove);
-                    exportItems.setStackInSlot(0, outputStack);
-                    this.itemsStoredInside -= amountOfItemsToRemove;
-                    if (this.itemsStoredInside == 0) {
-                        this.virtualItemStack = ItemStack.EMPTY;
-                    }
+                    GTTransferUtils.moveInventoryItems(itemInventory, exportItems);
 
                     markDirty();
                 }
-
             }
+
             if (isAutoOutputItems()) {
                 pushItemsIntoNearbyHandlers(currentOutputFacing);
             }
+
+            if (this.voiding && !importItems.getStackInSlot(0).isEmpty()) {
+                importItems.setStackInSlot(0, ItemStack.EMPTY);
+            }
+
             if (previousStack == null || !areItemStackIdentical(previousStack, virtualItemStack)) {
-                writeCustomData(UPDATE_ITEM, buf -> buf.writeItemStack(virtualItemStack));
+                writeCustomData(UPDATE_ITEM, buf -> {
+                    virtualItemStack.setCount(1);
```

Comment:
Why is changing the count necessary here? When does it ever become not `1`?

Reply:
i could also remove this as well